### ISSUE TITLE
Update LocalizedError to use more of the protocol and extend custom error type.

### ISSUE THREAD

**Issue by tfrank64** [IBM-Swift/BluePic#319]
This is a good starting point http://stackoverflow.com/a/39176551/2280737

**Comment by rfdickerson** [comment]
Here's another example:

```swift
enum TaskListError : LocalizedError {
    
    case noJSONinBody
    case noDescriptionInBody
    case descriptionTooShort(String)
    case descriptionTooLong(String)
    
    var errorDescription: String? {
        switch self {
        case .noJSONinBody: return "Must include JSON in the body"
        case .noDescriptionInBody: return "Must have the task description"
        case .descriptionTooShort(let string): return "\(string) too short"
        case .descriptionTooLong(let string): return "\(string) too long"
        }
    }
}

extension TaskListError: StringValuePairConvertible {
    var stringValuePairs: StringValuePair {
        return ["error": self.errorDescription ?? ""]
    }
}
```

**Comment by king33** [comment]
Using the example above for TaskListError, do we need a case for every instance that the LocalizedError is used?  Will it be possible for someone to define a unique error message without adding a new case to the enum?

`enum BluePicLocalizedError : LocalizedError {
    
    case noImagesByTag
    
    var errorDescription: String? {
        switch self {
        case .noImagesByTag: return "Failed to find images by tag." ...`

Both of the lines of code below compile-- is this legit?
`response.error = BluePicLocalizedError.noImagesByTag`
`response.error = BluePicLocalizedError(errorDescription: "Failed to retrieve all images.")
`

**Comment by rfdickerson** [comment]
I would do this instead:

```swift
//...
case noImagesByTag(String)

var errorDescription: String? {
    switch self {
    case .noImagesByTag(let tag): return "Failed to find images by tag \(tag)..."
```

You could add another case called `other` that could allow BluePic to throw some new error. Although, I'm kind of against this. I think a finite number of enumerable errors make for a more robust design.

**Comment by tfrank64** [comment]
If we do unique error cases, how many do you think there will be?

**Comment by king33** [comment]
There will be 13 cases with the current code base.

**Comment by tfrank64** [comment]
Hmm, if there is no way to consolidate some of them, then we should make 13 cases.

**Comment by king33** [comment]
Fixed by PR: #326
